### ISSUE TITLE
Use ToastUI as the markdown editor

### ISSUE THREAD

**Issue by ethanmdavidson** [BookStackApp/BookStack#1661]
**Describe the feature you'd like**
Use [ToastUI](https://ui.toast.com/tui-editor/) as the editor in Markdown mode. This has been mentioned in a few other issues (#842, #776) but I thought it should have its own issue specifically. 

**Describe the benefits this feature would bring to BookStack users**
Adding this feature would resolve several issues (#119, #842, #369, #315) and also provide additional features which have not even been requested yet (e.g. buttons/helpers in markdown mode). The features provided by ToastUI would directly support the BookStack project definition of "a pleasant and simple out of the box experience" while still providing "advanced power features to those that desire it". 

**Additional context**
Just a note: I have never used ToastUI before so I can't personally speak to it being a good choice. However, it is the one that seems to be mentioned the most around here, and the features it advertises seem to be a really strong fit for the BookStack use cases. If ToastUI is unacceptable for some reason, then I'm sure there is another markdown-based editor with WYSIWYG support that would fit this need ([awesome-wysiwyg](https://github.com/JefMari/awesome-wysiwyg)).

Also, as people have mentioned in other discussions of switching to a markdown-based editor, ToastUI (even in WYSIWYG mode) is limited to the formatting capabilities of markdown. However, this can be mitigated by leaving the existing WYSIWYG setting (with TinyMCE) as a site-wide option for people who need the full formatting power. People who don't need this full power can then use the Markdown setting (with ToastUI) for an easy markdown/WYSIWYG compatibility.

In other words, it would be set up like this:
BookStack can be run in two settings/modes: Markdown and WYSIWYG (For clarity, these should probably be renamed to "Simple Mode" and "Full Mode" or something like that).
If BookStack is in "Simple Mode", then the page editor will be ToastUI and all pages will be stored as markdown.
If BookStack is in "Full Mode", then the page editor will be TinyMCE and all pages will be stored in HTML (or w/e they are currently stored as).

**Comment by abulgatz** [comment]
From the demos, ToastUI certainly looks impressive, and offers features I've wished that BookStack had.

**Comment by ssddanbrown** [comment]
Thanks for the recommendation @ethanmdavidson. This is somewhat related to #1649 

I'm not really digging down into alternative editors right now, due to the time it takes to fully test each out to properly assess, but "Editor Alignment & Review" is on the roadmap to be looked at after the REST API. As part of that there will be a review of the editors themselves.

As with #1649 I'll keep this open for now but I'll likely merge it up into a wider editor-review issue/discussion in the future.

**Comment by ssddanbrown** [comment]
As mentioned above, Closing this in favour of #2738
